Model/Checkout/PayOrder.php

```diff
@@ -174,10 +156,9 @@ public function execute(int $orderId, string $amwalOrderId) : bool
             ->setLastRealOrderId($order->getIncrementId())
             ->setLastOrderStatus($order->getStatus());
 
-        $this->orderRepository->save($order);
+        $this->orderUpdate->update($order, $amwalOrderStatus, $historyComment);
+
         if($amwalOrderStatus == 'success') {
-            $this->invoiceAmwalOrder->execute($order, $amwalOrderData);
-            $this->setOrderUrl($order, $amwalOrderId);
```

Comment:
setOrderUrl has to be implemented in `OrderUpdate` as well. right?

Reply:
No, it's related to order details